File: Java/AbstractClassTest/src/SavingsAccount.java
package banking;

final class SavingsAccount extends Account {

	//final field value cannot be reassigned
	final static double MIN_BAL = 5000;

	SavingsAccount() {
		balance = MIN_BAL;
	}

	public void deposit(double amount) {
		balance += amount;
	}

	public void withdraw(double amount) throws InsufficientFundsException {
		if(balance - amount < MIN_BAL)
			throw new InsufficientFundsException();
		balance -= amount;
	}
}

File: DotNet/Inheritance/InterfaceTest1/Banking/SavingsAccount.cs
namespace Banking;

sealed class SavingsAccount : Account, IProfitable
{
    const decimal MinBal = 5000;

    internal SavingsAccount()
    {
        Balance = MinBal;
    }

    public override void Deposit(decimal amount)
    {
        Balance += amount;
    }

    public override void Withdraw(decimal amount)
    {
        if(Balance - amount < MinBal)
            throw new InsufficientFundsException();
        Balance -= amount;
    }

    public decimal AddInterest(int months)
    {
        decimal rate = Balance < 25000 ? IProfitable.MinRate : 4.5M;
        decimal interest = Balance * months * rate / 1200;
        Balance += interest;
        return interest;
    }
}
